source/node.go

```diff
@@ -177,10 +179,14 @@ func (ns *nodeSource) nodeAddresses(node *v1.Node) ([]string, error) {
 	var ipv6Addresses []string
 
```

Comment:
The Warning message is missing, with description that if users would like to expose private Node IPs, they should configure the flag explicitly `--expose-internal-ipv6`. Otherwise with next release, when we flip the flag, `external-dns` will silently update records for them.

With explicit test, we have a helper for that https://github.com/kubernetes-sigs/external-dns/blob/ecd57c86f526aebf1ecf54d0291de61ec3d48d6d/internal/testutils/log.go#L36